### PR TITLE
Fix using enum values in interfaces

### PR DESCRIPTION
**Pull request by marcindobry**
Fix for https://github.com/YousefED/typescript-json-schema/issues/122
***
Please:
- [x] Do not include the compiled `.js`, `.js.map`, or `.d.ts` in your pull request as it makes it harder to merge your changes. 
- [x] Make your pull request atomic, fixing one issue at a time unless there are many relevant issues that cannot be decoupled.
- [x] Provide a test case & update the documentation in the `Readme.md`

### CODE REVIEW

File: typescript-json-schema.ts [hunk 1]
```diff
@@ -372,7 +380,7 @@ export class JsonSchemaGenerator {
             }
         };
 
-        enm.members.forEach(member => {
+        members.forEach(member => {
```
Comment (domoritz) [inline typescript-json-schema.ts]: why not just `(node.kind === ts.SyntaxKind.EnumMember ? [enm] : enm.members).forEach(...)`?

File: typescript-json-schema.ts [hunk 2]
```diff
@@ -361,7 +361,15 @@ export class JsonSchemaGenerator {
     private getEnumDefinition(clazzType: ts.Type, tc: ts.TypeChecker, definition: Definition): Definition {
         const node = clazzType.getSymbol().getDeclarations()[0];
         const fullName = tc.typeToString(clazzType, undefined, ts.TypeFormatFlags.UseFullyQualifiedType);
-        const enm = <ts.EnumDeclaration>node;
+        let enm: ts.EnumDeclaration | ts.EnumMember;
```
Comment (domoritz) [inline typescript-json-schema.ts]: I think this can be const

**Comment by domoritz** [comment]
Thanks a lot for the test case and PR. Can you fix it so that the test pass?

File: typescript-json-schema.ts (hunk 1 above)
Comment (domoritz) [inline typescript-json-schema.ts]: See how I implemented it here: https://github.com/vega/typescript-to-json-schema/commit/462fe6d6a9ef7a52cd56edb8cfe930dd9f466a54. 

Btw, thanks for the PR. It helped me to also fix a bug in the other library I'm working on.

**Comment by marcindobry** [comment]
@domoritz Made the assignment more compact. 

The failing test is actually on master due to this change https://github.com/YousefED/typescript-json-schema/pull/119. I can make a separate PR to fix that, but I'm just unsure whether the tests are failing for a correct reason or just a mistake in defining the expected schema.

**Comment by domoritz** [comment]
Fantastic. Thanks!

**Pull request merged**